src/lib/gssapi/mechglue/g_init_sec_context.c

```diff
@@ -205,6 +208,19 @@ OM_uint32 *		time_rec;
 	    if (status != GSS_S_COMPLETE)
 		goto end;
 	}
+
+	if (used_req_flags == 0) {
+	    used_req_flags = union_ctx_id->req_flags;
+	    if (mech->gss_set_context_flags != NULL) {
+		status = mech->gss_set_context_flags(
```

Comment:
You're right that there is code to create the empty mech context, and can ignore this comment.